src/System.Text.Json/src/System/Text/Json/Document/JsonDocumentOptions.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.Json
+{
+    /// <summary>
+    /// Provides the ability for the user to define custom behavior when parsing JSON to create a <see cref="JsonDocument"/>.
+    /// </summary>
+    public struct JsonDocumentOptions
+    {
+        internal const int DefaultMaxDepth = 64;
+
+        private int _maxDepth;
+
+        /// <summary>
+        /// Defines how the <see cref="Utf8JsonReader"/> should handle comments when reading through the JSON.
+        /// By default <exception cref="JsonException"/> is thrown if a comment is encountered.
+        /// </summary>
+        public JsonCommentHandling CommentHandling { get; set; }
+
+        /// <summary>
+        /// Gets or sets the maximum depth allowed when reading JSON, with the default (i.e. 0) indicating a max depth of 64.
+        /// Reading past this depth will throw a <exception cref="JsonException"/>.
```

Comment:
This latter sentence seems like it should be in Remarks or in an Exception section.